flyteplugins/go/tasks/plugins/array/awsbatch/transformer.go

```diff
@@ -95,6 +95,8 @@ func FlyteTaskToBatchInput(ctx context.Context, tCtx pluginCore.TaskExecutionCon
 	if platformResources == nil {
 		platformResources = &v1.ResourceRequirements{}
 	}
+
+	flytek8s.SanitizeGPUResourceRequirements(res)
```

Comment:
This function is called internally in the `ApplyResourceOverrides` below I think. Are we then sanitizing in both locations?

Reply:
In the current state of the PR, the sanitation doesn't happen in `ApplyResourceOverrides` anymore but is moved to the new `SanitizeGPUResourceRequirements` which is called in `AddFlyteCustomizationsToContainer` directly before calling `ApplyResourceOverrides`.
This is why I think `SanitizeGPUResourceRequirements` would have to be called here as well.